components/motor/gpio/motor_encoder.go

```diff
@@ -163,13 +163,11 @@ type EncodedMotor struct {
 // EncodedMotorState is the core, non-statistical state for the motor.
 // Multiple values should be updated atomically at the same time.
 type EncodedMotorState struct {
-	regulated    bool
 	lastPowerPct float64
-	direction    float64
 }
 
 // rpmMonitor keeps track of the desired RPM and position.
-func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos float64) {
+func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos, direction float64) {
 	if m.encoder == nil {
 		panic("started rpmMonitor but have no encoder")
```

Comment:
Do we want panics in the code?

Reply:
I have no opinion on this. I can change it to a log if you'd prefer

Reply:
I don't think we're ever in this code if we don't have an encoder.

Reply:
if for some reason the encoder fails or something after the motor has already built, would that trigger a rebuild for motor? I'm not 100% sure what causes a rebuild

Reply:
so that'll be reflected in the .Position() call, which should return an error. If we've built or rebuilt, the constructor has already checked if this is nil and set it in the encoded motor struct.

Reply:
motor is an alwaysRebuild resource, so it will Close itself and call its constructor whenever is sees a change in the config.

Reply:
I'll just remove the check here